<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="22" hostname="DESKTOP-PO3T4V3" failures="32" tests="47" name="API Integration Automation Test" time="110.42" errors="0" timestamp="2022-10-31T16:50:53 AMT">
  <testcase classname="testCases.IqSoft_API_6_RollBackCreditWithDebit_Positive_Test" name="@BeforeClass setUp" time="13.368">
    <failure type="org.json.JSONException" message="A JSONObject text must begin with &amp;apos;{&amp;apos; at 1 [character 2 line 1]">
      <![CDATA[org.json.JSONException: A JSONObject text must begin with '{' at 1 [character 2 line 1]
at org.json.JSONTokener.syntaxError(JSONTokener.java:433)
at org.json.JSONObject.<init>(JSONObject.java:195)
at org.json.JSONObject.<init>(JSONObject.java:319)
at testCases.IqSoft_API_6_RollBackCreditWithDebit_Positive_Test.setUp(IqSoft_API_6_RollBackCreditWithDebit_Positive_Test.java:59)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 28 stack frames]]>
    </failure>
  </testcase> <!-- @BeforeClass setUp -->
  <testcase classname="testCases.IqSoft_API_2_Authorization_Positive_Test" name="AuthorizationAPIValidateStatusCod" time="0.061"/>
  <testcase classname="testCases.IqSoft_API_2_Authorization_Positive_Test" name="AuthorizationAPIValidatePositiveResponse" time="0.005"/>
  <testcase classname="testCases.IqSoft_API_3_GetBalance_Positive_Test" name="GetBalanceAPIValidateStatusCod" time="0.002"/>
  <testcase classname="testCases.IqSoft_API_3_GetBalance_Positive_Test" name="GetBalanceAPIValidatePositiveResponse" time="0.005"/>
  <testcase classname="testCases.IqSoft_API_4_Credit_Positive_Test" name="CreditAPIValidateStatusCod" time="0.003"/>
  <testcase classname="testCases.IqSoft_API_4_Credit_Positive_Test" name="CreditAPIValidatePositiveResponse" time="0.007">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Description:  expected [null] but found []">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Description:  expected [null] but found []
at testCases.IqSoft_API_4_Credit_Positive_Test.CreditAPIValidatePositiveResponse(IqSoft_API_4_Credit_Positive_Test.java:110)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidatePositiveResponse -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Positive_Test" name="DebitAPIValidateStatusCod" time="0.004"/>
  <testcase classname="testCases.IqSoft_API_5_Debit_Positive_Test" name="DebitAPIValidatePositiveResponse" time="0.002">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Description:  expected [null] but found [],
	expected [4.0] but found [5.0]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Description:  expected [null] but found [],
expected [4.0] but found [5.0]
at testCases.IqSoft_API_5_Debit_Positive_Test.DebitAPIValidatePositiveResponse(IqSoft_API_5_Debit_Positive_Test.java:121)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidatePositiveResponse -->
  <testcase classname="testCases.IqSoft_API_6_RollBackCreditWithDebit_Positive_Test" name="RollBackAPIValidateStatusCod" time="0.0">
    <skipped/>
  </testcase> <!-- RollBackAPIValidateStatusCod -->
  <testcase classname="testCases.IqSoft_API_6_RollBackCreditWithDebit_Positive_Test" name="RollBackAPIValidatePositiveResponse" time="0.0">
    <skipped/>
  </testcase> <!-- RollBackAPIValidatePositiveResponse -->
  <testcase classname="testCases.IqSoft_API_6_RollBackCreditWithDebit_Positive_Test" name="RollBackAPIValidatePositiveResponseBalanceCorrection" time="0.0">
    <skipped/>
  </testcase> <!-- RollBackAPIValidatePositiveResponseBalanceCorrection -->
  <testcase classname="testCases.IqSoft_API_6_RollBackCreditWithoutDebit_Positive_Test" name="RollBackAPIValidateStatusCod" time="0.003"/>
  <testcase classname="testCases.IqSoft_API_6_RollBackCreditWithoutDebit_Positive_Test" name="RollBackAPIValidatePositiveResponse" time="0.001">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Description:  expected [null] but found []">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Description:  expected [null] but found []
at testCases.IqSoft_API_6_RollBackCreditWithoutDebit_Positive_Test.RollBackAPIValidatePositiveResponse(IqSoft_API_6_RollBackCreditWithoutDebit_Positive_Test.java:82)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- RollBackAPIValidatePositiveResponse -->
  <testcase classname="testCases.IqSoft_API_6_RollBackCreditWithoutDebit_Positive_Test" name="RollBackAPIValidatePositiveResponseBalanceCorrection" time="0.002">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	afterRollBackCredit: 2.83 = beforeAll: 4.33 expected [4.33] but found [2.83]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
afterRollBackCredit: 2.83 = beforeAll: 4.33 expected [4.33] but found [2.83]
at testCases.IqSoft_API_6_RollBackCreditWithoutDebit_Positive_Test.RollBackAPIValidatePositiveResponseBalanceCorrection(IqSoft_API_6_RollBackCreditWithoutDebit_Positive_Test.java:96)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- RollBackAPIValidatePositiveResponseBalanceCorrection -->
  <testcase classname="testCases.IqSoft_API_6_RollBackDebit_Positive_Test" name="RollBackAPIValidateStatusCod" time="0.005"/>
  <testcase classname="testCases.IqSoft_API_6_RollBackDebit_Positive_Test" name="RollBackAPIValidatePositiveResponse" time="0.001">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Description:  expected [null] but found []">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Description:  expected [null] but found []
at testCases.IqSoft_API_6_RollBackDebit_Positive_Test.RollBackAPIValidatePositiveResponse(IqSoft_API_6_RollBackDebit_Positive_Test.java:94)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- RollBackAPIValidatePositiveResponse -->
  <testcase classname="testCases.IqSoft_API_6_RollBackDebit_Positive_Test" name="RollBackAPIValidatePositiveResponseBalanceCorrection" time="0.002">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	afterDebit: 7.98 = afterCredit: 1.83 + betAmountDebitConfig: 5.0 expected [6.83] but found [7.98],
	afterRollBackDebit: 1.98 = afterCredit: 1.83 expected [1.83] but found [1.98]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
afterDebit: 7.98 = afterCredit: 1.83 + betAmountDebitConfig: 5.0 expected [6.83] but found [7.98],
afterRollBackDebit: 1.98 = afterCredit: 1.83 expected [1.83] but found [1.98]
at testCases.IqSoft_API_6_RollBackDebit_Positive_Test.RollBackAPIValidatePositiveResponseBalanceCorrection(IqSoft_API_6_RollBackDebit_Positive_Test.java:110)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- RollBackAPIValidatePositiveResponseBalanceCorrection -->
  <testcase classname="testCases.IqSoft_API_2_Authorization_Negative_Test" name="AuthorizationAPIValidateResponseWithExpiredToken" time="0.3">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [29] but found [0],
	Error Description: null expected [SessionExpired] but found [null]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [29] but found [0],
Error Description: null expected [SessionExpired] but found [null]
at testCases.IqSoft_API_2_Authorization_Negative_Test.AuthorizationAPIValidateResponseWithExpiredToken(IqSoft_API_2_Authorization_Negative_Test.java:43)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- AuthorizationAPIValidateResponseWithExpiredToken -->
  <testcase classname="testCases.IqSoft_API_2_Authorization_Negative_Test" name="AuthorizationAPIValidateResponseWithInvalidToken" time="0.285">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [22] but found [0],
	Error Description: null expected [ClientNotFound] but found [null]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [22] but found [0],
Error Description: null expected [ClientNotFound] but found [null]
at testCases.IqSoft_API_2_Authorization_Negative_Test.AuthorizationAPIValidateResponseWithInvalidToken(IqSoft_API_2_Authorization_Negative_Test.java:71)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- AuthorizationAPIValidateResponseWithInvalidToken -->
  <testcase classname="testCases.IqSoft_API_2_Authorization_Negative_Test" name="AuthorizationAPIValidateResponseWithInvalidProductID" time="0.311">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [43] but found [0],
	Error Description: null expected [ProductNotFound] but found [null]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [43] but found [0],
Error Description: null expected [ProductNotFound] but found [null]
at testCases.IqSoft_API_2_Authorization_Negative_Test.AuthorizationAPIValidateResponseWithInvalidProductID(IqSoft_API_2_Authorization_Negative_Test.java:96)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- AuthorizationAPIValidateResponseWithInvalidProductID -->
  <testcase classname="testCases.IqSoft_API_3_GetBalance_Negative_Test" name="GetBalanceAPIValidateResponseWithExpiredToken" time="0.375">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [29] but found [0],
	Error Description: null expected [SessionExpired] but found [null]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [29] but found [0],
Error Description: null expected [SessionExpired] but found [null]
at testCases.IqSoft_API_3_GetBalance_Negative_Test.GetBalanceAPIValidateResponseWithExpiredToken(IqSoft_API_3_GetBalance_Negative_Test.java:41)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- GetBalanceAPIValidateResponseWithExpiredToken -->
  <testcase classname="testCases.IqSoft_API_3_GetBalance_Negative_Test" name="GetBalanceAPIValidateResponseWithInvalidToken" time="0.323">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [22] but found [0],
	Error Description: null expected [error login] but found [null]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [22] but found [0],
Error Description: null expected [error login] but found [null]
at testCases.IqSoft_API_3_GetBalance_Negative_Test.GetBalanceAPIValidateResponseWithInvalidToken(IqSoft_API_3_GetBalance_Negative_Test.java:65)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- GetBalanceAPIValidateResponseWithInvalidToken -->
  <testcase classname="testCases.IqSoft_API_3_GetBalance_Negative_Test" name="GetBalanceAPIValidateResponseWithInvalidCurrencyID" time="0.376">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [20] but found [0],
	Error Description: null expected [CurrencyNotExists] but found [null]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [20] but found [0],
Error Description: null expected [CurrencyNotExists] but found [null]
at testCases.IqSoft_API_3_GetBalance_Negative_Test.GetBalanceAPIValidateResponseWithInvalidCurrencyID(IqSoft_API_3_GetBalance_Negative_Test.java:89)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- GetBalanceAPIValidateResponseWithInvalidCurrencyID -->
  <testcase classname="testCases.IqSoft_API_3_GetBalance_Negative_Test" name="GetBalanceAPIValidateResponseWithOtherCurrencyID" time="0.67">
    <failure type="java.lang.AssertionError" message="balanceUSD VS balanceEUR did not expect [0.98] but found [0.98]">
      <![CDATA[java.lang.AssertionError: balanceUSD VS balanceEUR did not expect [0.98] but found [0.98]
at testCases.IqSoft_API_3_GetBalance_Negative_Test.GetBalanceAPIValidateResponseWithOtherCurrencyID(IqSoft_API_3_GetBalance_Negative_Test.java:113)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 34 stack frames]]>
    </failure>
  </testcase> <!-- GetBalanceAPIValidateResponseWithOtherCurrencyID -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithExpiredToken" time="1.061">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [29] but found [21],
	Error Description: Insufficient funds expected [SessionExpired] but found [Insufficient funds]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [29] but found [21],
Error Description: Insufficient funds expected [SessionExpired] but found [Insufficient funds]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithExpiredToken(IqSoft_API_4_Credit_Negative_Test.java:61)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithExpiredToken -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithInvalidToken" time="0.963">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [22] but found [21],
	Error Description: Insufficient funds expected [error login] but found [Insufficient funds]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [22] but found [21],
Error Description: Insufficient funds expected [error login] but found [Insufficient funds]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithInvalidToken(IqSoft_API_4_Credit_Negative_Test.java:102)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithInvalidToken -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithInvalidProductID" time="0.869">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [43] but found [21],
	Error Description: Insufficient funds expected [game not available] but found [Insufficient funds]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [43] but found [21],
Error Description: Insufficient funds expected [game not available] but found [Insufficient funds]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithInvalidProductID(IqSoft_API_4_Credit_Negative_Test.java:144)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithInvalidProductID -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseUsingTransactionIDTwice" time="1.532"/>
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseUsingAmountHigherThenBalance" time="0.879">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [63] but found [21],
	Error Description: Insufficient funds expected [balance less then bet Amount] but found [Insufficient funds]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [63] but found [21],
Error Description: Insufficient funds expected [balance less then bet Amount] but found [Insufficient funds]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseUsingAmountHigherThenBalance(IqSoft_API_4_Credit_Negative_Test.java:237)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseUsingAmountHigherThenBalance -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithInvalidAmount" time="0.888">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithInvalidAmount(IqSoft_API_4_Credit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithInvalidAmount" time="0.871">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	AmountBeforeCredit: 0.98 = amountAfterCredit : 1.98 expected [1.98] but found [0.98]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
AmountBeforeCredit: 0.98 = amountAfterCredit : 1.98 expected [1.98] but found [0.98]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithInvalidAmount(IqSoft_API_4_Credit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithInvalidAmount" time="0.982">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	AmountBeforeCredit: 1.98 = amountAfterCredit : 3.48 expected [3.48] but found [1.98]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
AmountBeforeCredit: 1.98 = amountAfterCredit : 3.48 expected [3.48] but found [1.98]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithInvalidAmount(IqSoft_API_4_Credit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithInvalidAmount" time="1.963">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	AmountBeforeCredit: 3.48 = amountAfterCredit : 4.53 expected [4.53] but found [3.48]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
AmountBeforeCredit: 3.48 = amountAfterCredit : 4.53 expected [4.53] but found [3.48]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithInvalidAmount(IqSoft_API_4_Credit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_4_Credit_Negative_Test" name="CreditAPIValidateResponseWithInvalidAmount" time="0.863">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	AmountBeforeCredit: 4.53 = amountAfterCredit : 5.54 expected [5.54] but found [4.53]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
AmountBeforeCredit: 4.53 = amountAfterCredit : 5.54 expected [5.54] but found [4.53]
at testCases.IqSoft_API_4_Credit_Negative_Test.CreditAPIValidateResponseWithInvalidAmount(IqSoft_API_4_Credit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- CreditAPIValidateResponseWithInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseWithInvalidToken" time="1.711">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [22] but found [0],
	Error Description:  expected [error login] but found [],
	amountAfterCredit = amountAfterDebit4.54 != 9.54 expected [9.54] but found [4.54]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [22] but found [0],
Error Description:  expected [error login] but found [],
amountAfterCredit = amountAfterDebit4.54 != 9.54 expected [9.54] but found [4.54]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseWithInvalidToken(IqSoft_API_5_Debit_Negative_Test.java:70)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseWithInvalidToken -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseUsingInvalidProductID" time="1.582">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 0 expected [43] but found [0],
	Error Description:  expected [game not available] but found [],
	amountAfterCredit = amountAfterDebit 8.54  =  13.54 expected [13.54] but found [8.54]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 0 expected [43] but found [0],
Error Description:  expected [game not available] but found [],
amountAfterCredit = amountAfterDebit 8.54  =  13.54 expected [13.54] but found [8.54]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseUsingInvalidProductID(IqSoft_API_5_Debit_Negative_Test.java:122)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseUsingInvalidProductID -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseUsingTransactionIDTwice" time="2.758">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [0] but found [21],
	Error Description: Transaction is already processed expected [null] but found [Transaction is already processed]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [0] but found [21],
Error Description: Transaction is already processed expected [null] but found [Transaction is already processed]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseUsingTransactionIDTwice(IqSoft_API_5_Debit_Negative_Test.java:194)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseUsingTransactionIDTwice -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseUsingInvalidCreditTransactionID" time="0.874">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [50] but found [21],
	Error Description: CreditTransactionId Not Found expected [CanNotConnectCreditAndDebit] but found [CreditTransactionId Not Found]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [50] but found [21],
Error Description: CreditTransactionId Not Found expected [CanNotConnectCreditAndDebit] but found [CreditTransactionId Not Found]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseUsingInvalidCreditTransactionID(IqSoft_API_5_Debit_Negative_Test.java:232)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseUsingInvalidCreditTransactionID -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseUsingInvalidAmount" time="1.473">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	expected [14.54] but found [15.54]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
expected [14.54] but found [15.54]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseUsingInvalidAmount(IqSoft_API_5_Debit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseUsingInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseUsingInvalidAmount" time="1.665">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	expected [12.04] but found [13.54]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
expected [12.04] but found [13.54]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseUsingInvalidAmount(IqSoft_API_5_Debit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseUsingInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseUsingInvalidAmount" time="1.556">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	expected [9.99] but found [11.04]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
expected [9.99] but found [11.04]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseUsingInvalidAmount(IqSoft_API_5_Debit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseUsingInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_5_Debit_Negative_Test" name="DebitAPIValidateResponseUsingInvalidAmount" time="1.523">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	ResponseCode: 21 expected [39] but found [21],
	Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
	expected [7.99] but found [8.99]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
ResponseCode: 21 expected [39] but found [21],
Error Description: Invalid request params expected [WrongOperationAmount] but found [Invalid request params],
expected [7.99] but found [8.99]
at testCases.IqSoft_API_5_Debit_Negative_Test.DebitAPIValidateResponseUsingInvalidAmount(IqSoft_API_5_Debit_Negative_Test.java:281)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- DebitAPIValidateResponseUsingInvalidAmount -->
  <testcase classname="testCases.IqSoft_API_6_RollBack_Negative_Test" name="RollBackAPIValidateResponseWithInvalidToken" time="12.209">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	amountAfterRollBack: 7.99 = amountAfterCredit: 6.99 expected [6.99] but found [7.99]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
amountAfterRollBack: 7.99 = amountAfterCredit: 6.99 expected [6.99] but found [7.99]
at testCases.IqSoft_API_6_RollBack_Negative_Test.RollBackAPIValidateResponseWithInvalidToken(IqSoft_API_6_RollBack_Negative_Test.java:67)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- RollBackAPIValidateResponseWithInvalidToken -->
  <testcase classname="testCases.IqSoft_API_6_RollBack_Negative_Test" name="RollBackAPIValidateResponseUsingInvalidProductID" time="12.068">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	amountAfterRollBack : 7.99 = amountAfterCredit: 6.99 expected [6.99] but found [7.99]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
amountAfterRollBack : 7.99 = amountAfterCredit: 6.99 expected [6.99] but found [7.99]
at testCases.IqSoft_API_6_RollBack_Negative_Test.RollBackAPIValidateResponseUsingInvalidProductID(IqSoft_API_6_RollBack_Negative_Test.java:118)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- RollBackAPIValidateResponseUsingInvalidProductID -->
  <testcase classname="testCases.IqSoft_API_6_RollBack_Negative_Test" name="RollBackAPIValidateResponseUsingSameTransactionIDForRollBackAndCredit" time="1.259"/>
  <testcase classname="testCases.IqSoft_API_6_RollBack_Negative_Test" name="RollBackAPIValidateResponseUsingSameCreditTransactionIDTwice" time="13.149"/>
  <testcase classname="testCases.IqSoft_API_6_RollBack_Negative_Test" name="RollBackAPIValidateResponseUsingSameRollBackTransactionIDTwice" time="13.39"/>
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
  <testcase classname="testCases.BaseTest" name="@AfterSuite tearDownSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @AfterSuite tearDownSuite -->
  <testcase classname="testCases.BaseTest" name="@BeforeSuite setupSuite" time="0.0">
    <ignored/>
  </testcase> <!-- @BeforeSuite setupSuite -->
</testsuite> <!-- API Integration Automation Test -->
